apps/server-monolith/app/models/users/authentication.rb

```diff
@@ -0,0 +1,148 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: users_authentications
+#
+#  id                     :bigint           not null, primary key
+#  confirmation_sent_at   :datetime
+#  confirmation_token     :string
+#  confirmed_at           :datetime
+#  current_sign_in_at     :datetime
+#  current_sign_in_ip     :string
+#  deleted_at             :datetime
+#  email                  :string
+#  encrypted_password     :string           default(""), not null
+#  failed_attempts        :integer          default(0), not null
+#  last_sign_in_at        :datetime
+#  last_sign_in_ip        :string
+#  locked_at              :datetime
+#  remember_created_at    :datetime
+#  reset_password_sent_at :datetime
+#  reset_password_token   :string
+#  sign_in_count          :integer          default(0), not null
+#  unconfirmed_email      :string
+#  unlock_token           :string
+#  created_at             :datetime         not null
+#  updated_at             :datetime         not null
+#  user_id                :bigint           not null
+#
+# Indexes
+#
+#  index_users_authentications_on_confirmation_token    (confirmation_token) UNIQUE
+#  index_users_authentications_on_deleted_at            (deleted_at)
+#  index_users_authentications_on_email                 (email) UNIQUE
+#  index_users_authentications_on_reset_password_token  (reset_password_token) UNIQUE
+#  index_users_authentications_on_unlock_token          (unlock_token) UNIQUE
+#  index_users_authentications_on_user_id               (user_id)
+#
+# Foreign Keys
+#
+#  fk_rails_...  (user_id => pods.id)
+#
+module Users
+  class Authentication < ApplicationRecord
+    belongs_to :user, optional: true
+
+    ## Devise
+    devise :database_authenticatable, :registerable, :lockable, :async,
+      :recoverable, :rememberable, :confirmable, :trackable, :omniauthable, :validatable
+
+    has_many :federated_identities, class_name: 'Users::AuthenticationFederatedIdentity',
+      foreign_key: :users_authentication_id, inverse_of: :authentication, dependent: :destroy
+
+    validates :username, presence: true, on: :create
+    validates :display_name, presence: true, on: :create
+
+    # Make devise happy
+    def name
+      user&.display_name || email
+    end
+
+    # Extra props on create
+    attr_accessor :omniauth_provider, :omniauth_uid
+
+    # Pass to user
+    attr_accessor :username, :display_name, :external_avatar_url
+
+    before_create :bind_or_create_user!
+    after_save :bind_federation_identity!
+
+    # Setup user_id before create
+    def bind_or_create_user!
+      self.user_id = User.find_or_create_by!({ username: username, display_name: display_name,
```

Comment:
change `bind_or_create_user!` to `create_user!`.

Since `federated_identities` now `belongs_to :authentication`, there is no case where the corresponding user already exists when the authentication is created.